source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesScriptServiceV1IntegrationTest.cs

```diff
@@ -1,40 +1,47 @@
+using System.Diagnostics;
 using FluentAssertions;
+using FluentAssertions.Execution;
 using Octopus.Tentacle.Client.Scripts.Models;
 using Octopus.Tentacle.Client.Scripts.Models.Builders;
 using Octopus.Tentacle.CommonTestUtils;
 using Octopus.Tentacle.CommonTestUtils.Diagnostics;
 using Octopus.Tentacle.Contracts;
 using Octopus.Tentacle.Contracts.Capabilities;
+using Octopus.Tentacle.Contracts.ClientServices;
 using Octopus.Tentacle.Contracts.KubernetesScriptServiceV1;
 using Octopus.Tentacle.Tests.Integration.Common.Builders.Decorators;
+using Octopus.Tentacle.Kubernetes.Tests.Integration.Util;
+using Octopus.Tentacle.Tests.Integration.Common.Builders.Decorators.Proxies;
 
 namespace Octopus.Tentacle.Kubernetes.Tests.Integration;
 
 [TestFixture]
 public class KubernetesScriptServiceV1IntegrationTest : KubernetesAgentIntegrationTest
 {
-    public KubernetesScriptServiceV1IntegrationTest()
+    IRecordedMethodUsages recordedMethodUsages = null!;
+
+    protected override TentacleServiceDecoratorBuilder ConfigureTentacleServiceDecoratorBuilder(TentacleServiceDecoratorBuilder builder)
     {
-        TentacleServiceDecoratorBuilder = new TentacleServiceDecoratorBuilder()
+        builder.RecordMethodUsages<IAsyncClientKubernetesScriptServiceV1>(out var recordedUsages)
             .DecorateCapabilitiesServiceV2With(d => d
-                .DecorateGetCapabilitiesWith((inner, options) => Task.FromResult(new CapabilitiesResponseV2(new List<string> { nameof(IFileTransferService), nameof(IKubernetesScriptServiceV1) }))));
+                .DecorateGetCapabilitiesWith((_, _) => Task.FromResult(new CapabilitiesResponseV2(new List<string> { nameof(IFileTransferService), nameof(IKubernetesScriptServiceV1) }))));
+
+        recordedMethodUsages = recordedUsages;
+
+        return builder;
     }
 
     [Test]
-    public async Task SimpleHelloWorld()
+    public async Task RunSimpleScript()
     {
         // Arrange
         var logs = new List<ProcessOutput>();
         var scriptCompleted = false;
-        string scriptBody = @"echo ""Hello World""
-for i in $(seq 1 50);
-do
-    echo $i
-    sleep 0.1
-done
-";
+
         var command = new ExecuteKubernetesScriptCommandBuilder(LoggingUtils.CurrentTestHash())
-            .WithScriptBody(scriptBody)
+            .WithScriptBody(script => script
+                .Print("Hello World")
+                .PrintNTimesWithDelay("Yep", 50, TimeSpan.FromMilliseconds(100)))
```

Comment:
Is it necessary to run the full 5 seconds? It can probably be shortened and still be a useful test.

Reply:
Yeah, that's a good call. We do want it to run long enough that we exercise the pod monitor etc, but I'll drop down to 3s